### DISCUSSION TITLE
spotlight seems to be running on the macos-latest runners and taking up a lot of CPU

### DISCUSSION THREAD

**Discussion by graingert** [actions/virtual-environments#5032] [General questions]
when debugging slow/flakey builds on macos I'm finding a process "mdworker-sizing" taking up a lot of CPU. This seems related to spotlight indexing:

I took the following screenshots using tmate on this CI action https://github.com/graingert/distributed/runs/5037116104?check_suite_focus=true
![image](https://user-images.githubusercontent.com/413772/152771151-0c2359df-ba99-47cf-bbbf-20883b37d4de.png)
![image](https://user-images.githubusercontent.com/413772/152771182-20a0d7e7-cf0c-4f08-b5d1-18a9ef984a89.png)

spotlight indexing appears to be enabled for `/` for the macos-latest runners.  Should this be disabled by default?

**Comment by al-cheb** [comment]
If we disable spotlight service by default it will break some customers - https://github.com/actions/virtual-environments/issues/1760

**Reply by graingert** [reply]
does it have to be enabled for all of `/`? could it be disabled on the worker user directory?

**Reply by miketimofeev** [reply]
We will consider partially disabling spotlight with a proper announcement in the repository but I'm afraid no ETA for now as we have tasks with higher priority.

**Comment by igorboskovic3** [comment]
@graingert
Investigation showed that after boot up sequence indeed cpu utilization is from 23%-45%, which is used for indexing of all files by Spotlight and it takes from 3-4 minutes approximately. In addition if we exclude everything from /Applications folder (except Xcode, which was excluded in a first place and we had an issue regarding that exclusion, as you can see [here](https://github.com/actions/virtual-environments/issues/1760)) from indexing we will have approximately 1 minute less from that process. I thing its not enough for customer to resolve that issue. Also we can encounter further errors from those exclusions as we already did on exclusion of Xcode.

**Reply by igorboskovic3** [reply]
I will try one more thing, to exclude everything, except  /Application and see if that can help

**Reply by assrat4inone** [reply]
Yes sir 1 minute is not enough, please 🙏 tell them to increase up to 2 minutes.

**Comment by samus-aran** [comment]
We're locking this discussion because it has not had recent activity and/or other members have asked for more information to assist you but received no response. Thank you for helping us maintain a productive and tidy community for all our members.
